enterprise/cmd/frontend/auth/githuboauth/provider.go

```diff
@@ -52,7 +59,13 @@ func parseProvider(p *schema.GitHubAuthProvider, sourceCfg schema.AuthProviders)
 			}, sessionKey),
 			nil,
 		),
-	}), nil
+	}), messages
+}
+
+var clientIDSecretValidator = regexp.MustCompile("^[a-z0-9]*$")
```

Comment:
Consider using [lazyregexp](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+lazyregexp&patternType=literal) which only compiles the regexp on the first use to reduce the app start up time.
```suggestion
var clientIDSecretValidator = lazyregexp.MustCompile("^[a-z0-9]*$")
```